src/scipp/core/binning.py

```diff
@@ -5,11 +5,11 @@
 from typing import Dict, List, Optional, Union, Sequence
 
 from .._scipp import core as _cpp
-from .variable import array, Variable, linspace, arange, epoch
+from .variable import array, Variable, linspace, arange, epoch, scalar
 from .math import round as round_
 
 
-def make_histogrammed(x: Union[_cpp.DataArray, _cpp.Dataset], *,
+def make_histogrammed(x: Union[_cpp.Variable, _cpp.DataArray, _cpp.Dataset], *,
                       edges: _cpp.Variable) -> Union[_cpp.DataArray, _cpp.Dataset]:
```

Comment:
To be perfectly precise, the return type should be tied to the type of x. E.g. using https://docs.python.org/3/library/typing.html#typing.overload
But I don't know how that is rendered in the docs and help messages. I'm leaving it up to you if you want to implement it.